addons/account/models/partner.py

```diff
@@ -489,6 +489,18 @@ def _onchange_company_id(self):
             company = self.env.company
         return {'domain': {'property_account_position_id': [('company_id', 'in', [company.id, False])]}}
 
+    @api.onchange('parent_id')
+    def onchange_parent_id(self):
+        result = super().onchange_parent_id()
+        if self.sudo()._origin.journal_item_count:
```

Comment:
But `journal_item_count` is not a stored field, which was the only reason to check `customer_rank`. Even worse, it is computed by doing a  `search_count` in a loop.

I don't think that that cost is worth for this issue.

Reply:
What about checking invoice_ids ?